### ISSUE TITLE
two questions about the paper and codes

### ISSUE THREAD

**Issue by yxgyxgyxg** [lushleaf/Structure-free-certified-NLP#2]
Dear author, I have a few questions about your paper and code. I hope you can find time to reply:
1. When training classifier F, I found that the code just fine-tuned Bert. Why didn't use the data augmentation mentioned in your paper?
2.I found that the Certifified Robustness method mentioned in your paper is only used in the testing stage. I feel that if this is the case, the classifier will not get any improvement in accuracy. When faced with carefully designed adversarial samples, the classifier will still make mistakes. Please point out my misunderstanding, thank you very much.
I look forward to your reply.

**Comment by lushleaf** [comment]
Hi, thanks very much for your questions!

For the first question: thanks for pointing this out! Sorry we have some bugs when cleaning up the code. Line 519 in train.py should call load_and_cache_examples_randomized to generate a perturbed training set for each epoch. And we should create a random_smooth similar to evaluation.py. For some reason we can't access to the server that has the code and we will update the repo once we have it.

For the second question: I am not very sure about your question. We calculate the certified accuracy during the test is because we should use test set to compared the model performance. It is possible that there can be a strong attack that make the model give wrong answer. But if an example is certified correct, that means no matter how we apply word substitution attack on it, the model is guaranteed to predict correctly.

How to improve the certified accuracy of the randomized model (i.e., make the model certified robust to more sentence) is an another research direction. We only consider the data augmentation for improving it. There can be many other ways.

**Comment by yxgyxgyxg** [comment]
Your papers and ideas are excellent. Thank you very much for your reply

**Comment by Hazelsuko07** [comment]
Dear authors,

Thanks for sharing the code! I was wondering if you have got a chance to access your server and update the repo? Thanks!

**Comment by lushleaf** [comment]
Hi, the repo has been updated. Thanks!

**Comment by Hazelsuko07** [comment]
Thanks for the update! 

However, I am not sure if the code is your latest version. 

Basically, your update loads randomized examples before starting training by calling the `load_and_cache_examples_randomized` function:
> https://github.com/lushleaf/Structure-free-certified-NLP/blob/c2b86959b1ed4022e44775c6e26cca3a462e1384/train.py#L519

But as defined, the function should take a `random_smooth` instance and an epoch number as input.

> https://github.com/lushleaf/Structure-free-certified-NLP/blob/c2b86959b1ed4022e44775c6e26cca3a462e1384/train.py#L258

I guess you may want to have this fixed. (Please let me know if I missed anything here.)

Thanks!

**Comment by lushleaf** [comment]
Hi, thanks for pointing this out. The repo is updated.

**Comment by Hazelsuko07** [comment]
Thanks for the quick update 👍
